includes/managers/icons/migrations.php

```diff
@@ -0,0 +1,298 @@
+			];
+		}
+
+		foreach ( self::get_migration_dictionary() as $mapping_by_version ) {
+			if ( isset( $mapping_by_version[ $value['value'] ] ) ) {
+				$value = $mapping_by_version[ $value['value'] ];
+			}
+		}
+
+		// Make sure there's no value in the old 'fa ' format, even if no replacement found.
+		if ( '' === $value['library'] ) {
+			$value = [
+				'value'   => 'fas ' . str_replace( 'fa ', '', $value ),
+				'library' => 'fa-solid',
+			];
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Get the migration dictionary (mapping of the icon changes) by versions.
+	 *
+	 * @return array
+	 */
+	public static function get_migration_dictionary() {
+		static $migration_dictionary = [];
+
+		if ( ! empty( $migration_dictionary ) ) {
+			return $migration_dictionary;
+		}
+
+		$current_version = Icons_Manager::get_current_fa_version();
+		$oldest_legacy_version = 4;
+
+		for ( $i = $oldest_legacy_version; $i <= $current_version; $i++ ) {
+			$mapping_by_version = sprintf( 'mapping-v%s-to-v%s', $i, $i + 1 );
+			$mapping_file = ELEMENTOR_ASSETS_PATH . "/lib/font-awesome/migration/$mapping_by_version.js";
+
+			if ( file_exists( $mapping_file ) ) {
+				$migration_dictionary[] = json_decode( Utils::file_get_contents( $mapping_file ), true );
+			}
+		}
+
+		return $migration_dictionary;
+	}
+
+	/**
+	 * On import, migrate the icon values.
+	 *
+	 * @param array $element        settings array
+	 * @param string $old_control   old control id
+	 * @param string $new_control   new control id
+	 * @param bool $remove_old      boolean weather to remove old control or not
+	 *
+	 * @return array
+	 */
+	public static function on_import_migration( array $element, $old_control = '', $new_control = '', $remove_old = false ) {
+		if ( ! isset( $element['settings'][ $old_control ] ) || isset( $element['settings'][ $new_control ] ) ) {
+			return $element;
+		}
+
+		// Case when old value is saved as empty string
+		$new_value = [
+			'value' => '',
+			'library' => '',
+		];
+
+		// Case when old value needs migration
+		if ( ! empty( $element['settings'][ $old_control ] ) && self::is_migration_required() ) {
+			$new_value = self::fa_icon_value_migration( $element['settings'][ $old_control ] );
+		}
+
+		$element['settings'][ $new_control ] = $new_value;
+
+		//remove old value
+		if ( $remove_old ) {
+			unset( $element['settings'][ $old_control ] );
+		}
+
+		return $element;
+	}
+
+	/**
+	 * Is an icon migration required in the site content?
+	 *
+	 * @since 3.19.0
+	 * @return bool
+	 */
+	public static function is_migration_required() {
+		static $migration_required = false;
+
+		if ( false === $migration_required ) {
+			if ( empty( key( Upgrade_Manager::get_installs_history() ) ) ) {
+				// If `get_installs_history()` is empty, Elementor was never installed on this site.
+				update_option( 'elementor_' . self::LATEST_MIGRATION_OPTION, self::LATEST_MIGRATION_REQUIRED_VERSION );
+			} else {
+				$needs_upgrade = self::get_needs_upgrade_option();
```

Comment:
Maybe if the value is 'yes' then just return it here instead of continuing to the next checks.
+ Not sure that I understand the part below, let's talk about it.

Reply:
Done.